app/models/location_aware_booking_requests.rb

```diff
@@ -0,0 +1,21 @@
+class LocationAwareBookingRequests < SimpleDelegator
+  def initialize(booking_requests, booking_location)
+    @booking_location = booking_location
+    super(booking_requests)
+  end
+
+  def each
+    return enum_for(:each) unless block_given?
+
+    __getobj__.each do |booking_request|
+      yield LocationAwareBookingRequest.new(
+        booking_location: booking_location,
+        booking_request: booking_request
+      )
+    end
```

Comment:
The more I look at this the more I think it is more complex than needed.. 

We have turned an array of objects into a first class object then used a delegator to modify the each behaviour. 

Why not just map the array that way you don't need the SimpleDelegator?